source/material_model/rheology/visco_plastic.cc

```diff
@@ -294,6 +294,16 @@ namespace aspect
 
             // Step 5: limit the viscosity with specified minimum and maximum bounds
             output_parameters.composition_viscosities[j] = std::min(std::max(viscosity_yield, min_visc), max_visc);
+
+            // Step 6: If the composite rheological formulation is selected, recalculate the viscosity
+            std::vector<double> partial_strain_rates(5, 0.);
+            if (use_composite_rheology)
```

Comment:
@bobmyhill - Should we have a much larger if-else statement that skips over steps 1-5 if this option is enabled?

Reply:
Yes, I think the methods should be entirely separate; that way we can implement improvements/fix bugs in either one without affecting the other.

Reply:
@bobmyhill - After a bit of thought, one disadvantage to the latter approach (large if-else statement) is that the compositing could still be used for the creep mechanisms in combination with plasticity.

In this scenario, the iterative procedure could be used instead for steps 1a-c to calculate the bulk viscosity for all of the creep mechanisms. If the viscous stress is greater than the yield stress, then the viscosity is rescaled to the yield stress and the strain-rate is entirely taken up by plasticity.

However, maybe this is will lead excessive confusion about how the compositing is being used? It would also require quite a few internal checks for how parameters are being used in combination with each other. 

Perhaps best to keep things simple for now per your recommendation (i.e., one large if-else statement)? We can always add more complexity later on.

Reply:
I think keeping things as simple as possible is a good idea. I understand your point about using composite as composition creep only, but I don't think that will give the user any additional benefits, and as you say, it'll be harder to understand/explain how the compositing and various stabilization schemes work together.